feat: add event logs for vault config updates

hrharder opened the pull request:
fixes: #128 

Adds granular events for the following changes to a vaults parameters/configurations:

- ERC20 name set/updated
- ERC20 symbol set/updated
- Guest list set/updated
- Rewards address set/updated
- Deposit limit set/updated
- Performance fee set/updated
- Management fee set/updated
- Guardian address set/updated
- Emergency shutdown state set (not logged in initialization)
- Withdrawal queue set/updated

Are tests necessary for something this simple? Any feedback/requests on naming or indexed parameters or anything like that? First time using vyper as so appreciate any possible improvements.

Also -- to confirm, no linter for vyper? Followed pattern of two newlines after event declarations.

h/t @fubuloubu

fubuloubu commented on contracts/Vault.vy:
```diff
@@ -116,6 +116,46 @@ event StrategyReported:
     debtLimit: uint256
 
 
+event SetName:
+    name: String[64] # New active Vault ERC20 name
+
+
+event SetSymbol:
+    symbol: String[32] # New active Vault ERC20 symbol
+
+
```
Probably don't need these, should only really be used once or twice

fubuloubu commented on contracts/Vault.vy:
```diff
@@ -207,6 +247,13 @@ def __init__(
             convert(self, bytes32)
         )
     )
+    log SetName(self.name)
+    log SetSymbol(self.symbol)
+    log SetRewards(self.rewards)
+    log SetGuardian(self.guardian)
+    log SetPerformanceFee(self.performanceFee)
+    log SetManagementFee(self.managementFee)
+    log SetDepositLimit(self.depositLimit)
```
Use the input values/literals/constants for these, saving like ~1950 gas each or so

Also, put each one after where it's set

hrharder replied:
cool -- didn't know this re. gas

fubuloubu commented on contracts/Vault.vy:
```diff
@@ -116,6 +116,46 @@ event StrategyReported:
+    symbol: String[32] # New active Vault ERC20 symbol
+
+
+event SetGuestList:
+    guestList: indexed(address) # Vault guest list address
+
+
+event SetRewards:
+    rewards: indexed(address) # New active rewards recipient 
+
+
+event SetDepositLimit:
+    depositLimit: indexed(uint256) # New active deposit limit
+
+
+event SetPerformanceFee:
+    performanceFee: indexed(uint256) # New active performance fee
+
+
+event SetManagementFee:
+    managementFee: indexed(uint256) # New active management fee
+
+
+event SetGuardian:
+    guardian: indexed(address) # Address of the active guardian
```
No need to index these

Only index `strategy` address in `UpdateStrategy*` events

fubuloubu reviewed (changes requested):
Broadly, could use you use `Update*` instead? I know it's a bit more chars, but is more meaningful. Unless you have another suggestion. `Set*` doesn't see right to me.

hrharder commented:
@fubuloubu addressed comments in [a56619d](https://github.com/iearn-finance/yearn-vaults/pull/131/commits/a56619d372c3b6316bfe85cf787466e1b3e61d93) and added events for strategy param updates -- thank you

fubuloubu commented on contracts/Vault.vy:
```diff
@@ -116,6 +116,70 @@ event StrategyReported:
+    strategy: indexed(address) # Address of the strategy that is revoked
+
+
+event StrategyRemoved:
+    strategy: indexed(address) # Address of the strategy that is removed
```
```suggestion
event StrategyRemovedFromQueue:
    strategy: indexed(address) # Address of the strategy that is removed from the withdrawal queue
```

Should also have `StrategyAddedToQueue`?

fubuloubu commented on contracts/Vault.vy:
```diff
@@ -116,6 +116,70 @@ event StrategyReported:
+    performanceFee: uint256 # The new performance fee for the strategy
+
+
+event UpdateStrategy:
```
Change to `StrategyMigrated`

fubuloubu commented on contracts/Vault.vy:
```diff
@@ -116,6 +116,70 @@ event StrategyReported:
+    queue: address[MAXIMUM_STRATEGIES] # New active withdrawal queue
+
+
+event UpdateStrategyDebtLimit:
+    strategy: indexed(address) # Address of the strategy for the debt limit adjustment
+    debtLimit: uint256 # The new debt limit for the strategy
+
+
+event UpdateStrategyRateLimit:
+    strategy: indexed(address) # Address of the strategy for the rate limit adjustment
+    rateLimit: uint256 # The new rate limit for the strategy
+
+
+event UpdateStrategyPerformanceFee:
+    strategy: indexed(address) # Address of the strategy for the performance fee adjustment
+    performanceFee: uint256 # The new performance fee for the strategy
```
Maybe these could be `StrategyUpdate*` instead? To match the rest

fubuloubu commented on contracts/Vault.vy:
```diff
@@ -116,6 +116,70 @@ event StrategyReported:
 
 
+event UpdateGovernance:
+    governance: indexed(address) # New active governance
```
Not sure if this needs to be indexed

fubuloubu reviewed (approved):
A couple nitpicks, but overall looking good!

hrharder commented:
updated

fubuloubu commented:
@hrharder good to go? lgtm

hrharder commented:
@fubuloubu indeed it is -- send it, at your leisure
